sysdump/sysdump.go

```diff
@@ -738,11 +760,10 @@ func (c *Collector) Run() error {
 			Description: "Collecting the Cilium configuration",
 			Quick:       true,
 			Task: func(ctx context.Context) error {
-				v, err := c.Client.GetConfigMap(ctx, c.Options.CiliumNamespace, ciliumConfigMapName, metav1.GetOptions{})
-				if err != nil {
-					return fmt.Errorf("failed to collect the Cilium configuration: %w", err)
+				if c.CiliumConfigMap == nil {
+					return nil
```

Comment:
Should we return an error here as we did prior to the change if the cache cm is nil?

Reply:
in general i'm trying to avoid raising errors. the reason is that the sysdump is also used to collect tetragon-related resources. in that case it is expected that cilium-config configmap is not found.

note that we still log when sysdump fails to get cilium-config configmap in https://github.com/cilium/cilium-cli/pull/1992/files#diff-e9c40a8e7876ce0237b03aba3f2b4f81ad70f07ae0b8ff1cffa57e76ef5eec8fR260-R265